packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx

```diff
@@ -207,34 +216,90 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
     }
 
     private entryToTreeNode(entry: OutputDescriptor, idStringToNodeId: { [key: string]: number }): TreeNode {
-        const labels = [entry.name];
-        let tooltips = undefined;
-        if (entry.description) {
-            tooltips = [entry.description];
-        }
-        let id = idStringToNodeId[entry.id];
-        if (id === undefined) {
-            id = this._idGenerator++;
-            idStringToNodeId[entry.id] = id;
-        }
+        let id = idStringToNodeId[entry.id] ?? (idStringToNodeId[entry.id] = this._idGenerator++);
+
         let parentId = -1;
         if (entry.parentId) {
-            const existingId = idStringToNodeId[entry.parentId];
-            if (existingId === undefined) {
-                parentId = this._idGenerator++;
-                idStringToNodeId[entry.parentId] = parentId;
-            } else {
-                parentId = existingId;
-            }
+            parentId = idStringToNodeId[entry.parentId] ?? (idStringToNodeId[entry.parentId] = this._idGenerator++);
         }
-        return {
-            labels: labels,
-            tooltips: tooltips,
+
+        const treeNode: TreeNode = {
+            labels: [entry.name],
+            tooltips: entry.description ? [entry.description] : undefined,
             showTooltip: true,
             isRoot: false,
-            id: id,
-            parentId: parentId,
-            children: []
-        } as TreeNode;
+            id,
+            parentId,
+            children: [],
+            getEnrichedContent: this.createEnrichedContent(entry)
+        };
+
+        return treeNode;
     }
+
+    private createEnrichedContent(entry: OutputDescriptor): (() => JSX.Element) | undefined {
+        const isCustomizable = entry.capabilities?.canCreate === true;
+        const isDeletable = entry.capabilities?.canDelete === true;
+
+        if (!isCustomizable && !isDeletable) {
+            return undefined;
+        }
+
+        const nameSpanStyle = {
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            whiteSpace: 'nowrap',
+            minWidth: 0,
+            flexShrink: 1
+        };
+
+        if (isCustomizable) {
+            return (): JSX.Element => (
+                <>
+                    <span style={nameSpanStyle}>{entry.name}</span>
+                    <div className="remove-output-button-container" title={`Add custom analysis to ${entry.name}`}>
```

Comment:
I'm not sure if 'Add custom analysis to...' would apply in all cases. To be discussed what should be the button title.

Reply:
We are suggestion just 'Add custom view...', because this is in the (Available) Views panel.
We also suggest not to add the configurable output name because it might not combine well to the string in all cases, and it's visible next to the button anyways.